programs/CMakeLists.txt

```diff
@@ -30,49 +30,42 @@ endif()
 configure_file(${PROJECT_SOURCE_DIR}/programs/config.h.in 
                ${PROJECT_SOURCE_DIR}/programs/config.h)
 
-set(TEST_PROGRAMS_DEFINITIONS _POSIX_C_SOURCE=200809L
-                              _FILE_OFFSET_BITS=64
-                              HAVE_CONFIG_H)
-
-if (DEFLATE_TEST_SUPPORT__DO_NOT_USE)
-  list(APPEND TEST_PROGRAMS_DEFINITIONS TEST_SUPPORT__DO_NOT_USE)
-endif()
-
-if (BUILD_SHARED_LIBS)
-  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-endif()
-
 # Small library for common code
-add_library(programs)
-target_sources(programs PRIVATE prog_util.c test_util.c tgetopt.c)
-target_compile_definitions(programs PRIVATE ${TEST_PROGRAMS_DEFINITIONS})
-target_link_libraries(programs deflate)
+add_library(libdeflate_programs_common STATIC prog_util.c tgetopt.c)
```

Comment:
ok so it doesn't build on Windows (msvc & mingw), there is probably some missing logic here which are in Makefile & Makefile.msvc